modules/server_new/src/main/scala/observe/server/odb/IdTrackerOps.scala

```diff
@@ -68,12 +68,7 @@ trait IdTrackerOps[F[_]: MonadThrow](idTracker: Ref[F, ObsRecordedIds]):
         .at(obsId)
         .some
         .andThen(RecordedVisit.step)
-        .modify:
-          case Some(staleStepId) if stepId.isDefined =>
-            throw ObserveFailure.Unexpected:
-              s"Attempted to set current stepId for [$obsId] when it was already set. " +
-                s"Existing value [$staleStepId], new attempted value [${stepId.get}]"
-          case _                                     => stepId.map(RecordedStep(_))
+        .replace(stepId.map(RecordedStep(_)))
```

Comment:
Should we keep the safeguard and unset the current Step.Id on failure?

Reply:
I do unset the current Step.Id on failure, so yes, it should work even with the safeguard in place.